v03_pipeline/lib/reference_data/clinvar.py

```diff
@@ -150,3 +152,37 @@ def _parse_clinvar_release_date(local_vcf_path: str) -> str:
                 return None
 
     return None
+
+
+def download_and_import_clinvar_txt_file(url: str, types: dict, hl_filter: str | None):
+    with tempfile.NamedTemporaryFile(
+        suffix='.txt.gz',
+        delete=False,
+    ) as tmp_file:
+        urllib.request.urlretrieve(url, tmp_file.name)  # noqa: S310
+        return hl.import_table(
+            tmp_file.name,
+            force=True,
+            filter=hl_filter,
+            impute=True,
+            types=types,
+            missing='-',
+            min_partitions=3,  # recommended 2-4 partitions per core
+        )
+
+
+def download_and_import_clinvar_submission_summary() -> hl.Table:
+    # https://ftp.ncbi.nlm.nih.gov/pub/clinvar/tab_delimited/README - submission_summary.txt
+    logger.info('Getting clinvar submission summary')
+    ht = download_and_import_clinvar_txt_file(
+        url=CLINVAR_SUBMISSION_SUMMARY_URL,
+        types={'#VariationID': hl.tstr},
```

Comment:
We can just slot these on `hl.import_table` right?  Also `hl.tstr` is the default so the type isn't necessary.

Reply:
1. I split the download into two functions because I was originally downloading 2 files (the variant_summary.txt.gz file as well, but we can actually get all the info from submission_summary). I want to keep these two functions separate because in tests I can mock the function that calls `import_table` but still execute the hail code that does the selecting and aggregating of submitters and conditions.

2. iirc (thank you for this acronym) the `impute=True` in `import_table` would impute the `#VariationID` as an integer, but I coerce it to a string to join on the table from the clinvar vcf

Reply:
👍 to keeping the functions separate.  what about moving the args given to `download_and_import_clinvar_txt_file` to `hl.import_table` though?  

and yes, that looks right about the `impute=True`!  Could we just pass `{'#VariationID': hl.tstr, 'Submitter': hl.tstr, 'ReportedPhenotypeInfo': hl.str} instead of `impute`?  I think it's faster!

Reply:
moved the args to `import_table`, got rid of `impute` and passed the types!